### ISSUE TITLE
Batch Gradient Descent vs. Mini-Batch Gradient Descent

### ISSUE THREAD

**Issue by vis-yyin** [xqding/PEVAE_Paper#5]
Hi, I noticed that training in the **pfam_msa** folder and **simulated_msa** folder used different types of gradient descent: The elbo calculation, `loss.backward()` and `optimizer.step()` were done on the whole data set in **pfam_msa** train.py and on the mini-batches of data in **simulated_msa** train.py. Is there any specific reason for that?

Meanwhile, if I don't want to do mini-batch gradient descent but the data is really big, is it ok to calculate elbo on mini-batch of data in a for loop and then sum them up?

Thanks!
